src/execution_plan/execution_plan_build/convert_clauses.c

```diff
@@ -0,0 +1,243 @@
+#include "../execution_plan.h"
+#include "../../RG.h"
+#include "../ops/ops.h"
+#include "../../query_ctx.h"
+#include "../../util/rax_extensions.h"
+#include "../../ast/ast_build_ar_exp.h"
+#include "../ops/shared/scan_functions.h"
+#include "../../ast/ast_build_op_contexts.h"
+
+/* Reduces a filter operation into an apply operation. */
+void ExecutionPlan_ReduceFilterToApply(ExecutionPlan *plan, OpFilter *filter);
+
+/* Clause conversion logic in other files. */
+void buildMatchOpTree(ExecutionPlan *plan, AST *ast, const cypher_astnode_t *clause);
+void buildReturnOps(ExecutionPlan *plan, const cypher_astnode_t *clause);
+void buildWithOps(ExecutionPlan *plan, const cypher_astnode_t *clause);
+void buildMergeOp(ExecutionPlan *plan, AST *ast, const cypher_astnode_t *clause, GraphContext *gc);
```

Comment:
I prefer moving these to a header file, including `ExecutionPlanSegment_ConvertClause`